### PR TITLE
planner: fix bug of mpp wrongly set schema of exchanger

### PR DESCRIPTION
**Pull request by hanfei1991**
### What problem does this PR solve?

Problem Summary:
TiDB should set the right notnull flag for physical outer join's schema.
It has been fixed in Logical join but yet in Phyiscal join. This pr just port the fix. 
fix https://github.com/pingcap/tics/issues/1441

### What is changed and how it works?

What's Changed:
reset not null flag if the physical plan is outer join.

### Check List 

Tests 

- Unit test

### Release note 

- planner: fix bug of mpp wrongly set schema of exchanger

### CODE REVIEW

File: store/mockstore/unistore/cophandler/mpp.go [hunk 1]
```diff
@@ -148,6 +150,25 @@ func (b *mppExecBuilder) buildMPPJoin(pb *tipb.Join, children []*tipb.Executor)
...
+	} else if pb.JoinType == tipb.JoinType_TypeRightOuterJoin {
+		for _, tp := range leftCh.getFieldTypes() {
+			tp.Flag &= ^mysql.NotNullFlag
+		}
+		defaultInner := chunk.MutRowFromTypes(leftCh.getFieldTypes())
+		for i := range leftCh.getFieldTypes() {
+			defaultInner.SetDatum(i, types.NewDatum(nil))
+		}
+		e.defaultInner = defaultInner.ToRow()
+	}
```
Comment (fzhedu) [inline store/mockstore/unistore/cophandler/mpp.go:171]: why here need the default inner nil value? but the inner should not be nil.

**Comment by fzhedu** [comment]
could you add some tests on [mock] tiflash items of the cop and mpp?

**Comment by hanfei1991** [comment]
@fzhedu of course, but have to wait for https://github.com/pingcap/tidb/pull/22803 being merged, right?

**Comment by fzhedu** [comment]
can this PR  solve https://github.com/pingcap/tics/issues/1441?

**Comment by hanfei1991** [comment]
@fzhedu of course

**Comment by fzhedu** [comment]
> @fzhedu of course

I mean including this https://github.com/pingcap/tics/issues/1441#issuecomment-787838412

**Comment by hanfei1991** [comment]
@fzhedu ya, it does not solve it.

**Review by qw4990 — APPROVED** [review]

**Pull request merged**